src/features/events/store.ts

```diff
@@ -222,11 +239,58 @@ const eventsSlice = createSlice({
     eventsSelected: (state, action: PayloadAction<ZetkinEvent[]>) => {
       const toggledEvents = action.payload;
 
-      const uniqueEvents = new Set([
-        ...state.selectedEvents,
+      const uniqueEventIds = new Set([
+        ...state.selectedEventIds,
         ...toggledEvents.map((filtered) => filtered.id),
       ]);
-      state.selectedEvents = Array.from(uniqueEvents);
+      state.selectedEventIds = Array.from(uniqueEventIds);
+    },
+    eventsUpdate: (state, action: PayloadAction<[number[], string[]]>) => {
+      const [eventIds, mutating] = action.payload;
+      const items = state.eventList.items.filter((item) =>
+        eventIds.find((eventId) => item.id == eventId)
+      );
+
+      items.map((item) => {
+        item.mutating = mutating;
+        if (item.data) {
+          const event = item.data;
+          const dateStr = item.data.start_time.slice(0, 10);
+          state.eventsByDate[dateStr].items.map((i) => {
+            if (i.id === event.id) {
+              i.mutating = mutating;
+            }
+          });
+        }
+      });
+    },
+    eventsUpdated: (state, action: PayloadAction<ZetkinEvent[]>) => {
+      const updatedEvents = action.payload;
+      const items = state.eventList.items.filter((item) =>
+        updatedEvents.find((event) => item.id == event.id)
+      );
+
+      items.map((item) => {
```

Comment:
This could be an `items.forEach()` because we don't care about the return value, but no need to change this now as this will need to be refactored anyway when `eventsByDate` is modded to only include IDs.